numpy/_core/getlimits.py

```diff
@@ -572,6 +577,33 @@ def _init(self, dtype):
         self._str_smallest_subnormal = machar._str_smallest_subnormal.strip()
         return self
 
+    def _init_from_finfo_obj(self, dtype, finfo_obj):
+        self.dtype = numeric.dtype(dtype)
+        self.precision = finfo_obj.get('precision')
```

Comment:
I'd tend to make this a loop, as is done in `_init`:
```
for attr in ("precision", ....):
    setattr(self, attr, finfo.get(attr))
```
Though it does suggest that perhaps `finfo` should be something that has attributes instead of a mapping... (again like is done in `_init` -- indeed, with a bit of refactoring of `_get_machar`, it may be possible to factor out some of the `_init` code for and use it here).